### ISSUE TITLE
Is incremental learning possible?

### ISSUE THREAD

**Issue by shubzk** [ultralytics/ultralytics#9881]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have trained the model on a dataset of 5000 images. I get new 500 images every couple of days (same object classes). Is it possible to use the previous 'best.pt' weight file and train the model on just the new images to improve model performance?
NOTE: Object classes stay the same, there is just new data coming in continuously.

**Comment by glenn-jocher** [comment]
@shubzk yes, incremental learning is possible and a great way to keep improving your model with new data without starting from scratch. Here's how you can do it with the new batch of images:

1. Continue training using your previously saved `best.pt` file as the starting model.
2. Use the new dataset (or combined with the previous, depending on your strategy) for training.

Here's a concise example of how to do it:

```bash
yolo train model=path/to/best.pt data=new_data.yaml
```

Make sure `new_data.yaml` points to your new dataset (or the combined one). This will allow the model to learn from the new images incrementally. Just remember to regularly validate your model to monitor the performance on your validation set! 😊

**Comment by Kayzwer** [comment]
@shubzk remember to set `warmup_epoch` to 0, as it will use aggressive lr in few initial epochs
